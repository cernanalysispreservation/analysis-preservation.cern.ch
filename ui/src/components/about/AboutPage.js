import React from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';

import Box from 'grommet/components/Box';
import Heading from 'grommet/components/Heading';
import Section from 'grommet/components/Section';

class AboutPage extends React.Component {
  render() {
    return (
      <Box flex={true}>
        <Box flex={true} colorIndex="neutral-1-a" justify="center" align="center">
          <Section>
            <Box size="large">
              <Heading tag="h2">About Page</Heading>
            </Box>
          </Section>
        </Box>
      </Box>
    );
  }
}

AboutPage.propTypes = {
};

export default AboutPage;
