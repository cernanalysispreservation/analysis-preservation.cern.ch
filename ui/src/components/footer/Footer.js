import React from "react";
import Box from "grommet/components/Box";
import Footer from "grommet/components/Footer";
import Paragraph from "grommet/components/Paragraph";
import Menu from "grommet/components/Menu";
import Anchor from "grommet/components/Anchor";

class GrommetFooter extends React.Component {
  render() {
    return (
      <Footer size="small" justify="center" colorIndex="neutral-1" pad="small">
        <Box
          direction="row"
          align="center"
          justify="center"
          pad={{ between: "medium" }}
        >
          <Paragraph margin="none">
            Copyright 2018 Â© CERN. Created & Hosted by CERN. Powered by Invenio
            Software.
          </Paragraph>
          <Box
            direction="row"
            responsive={false}
            size="small"
            justify="between"
          >
            <Anchor
              label="Contact"
              href="mailto:analysis-preservation-support@cern.ch"
            />
            <Anchor label="About" path="/about" />
            <Anchor label="Status" path="/status" />
          </Box>
        </Box>
      </Footer>
    );
  }
}

export default GrommetFooter;
