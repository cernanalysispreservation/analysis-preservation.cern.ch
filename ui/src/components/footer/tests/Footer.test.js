import React from "react";
import { configure, shallow, render, mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import Footer from "../Footer";
import AboutPage from "../../about/AboutPage";
import StatusPage from "../../status/StatusPage";
import { MemoryRouter as Router, Route, Switch } from "react-router-dom";
import { Provider } from "react-redux";
import store from "../../../store/configureStore";

configure({ adapter: new Adapter() });

let wrapper;
beforeEach(() => {
  wrapper = mount(
    <Provider store={store}>
      <Router>
        <Switch>
          <Route path="/about" component={AboutPage} />
          <Route path="/status" component={StatusPage} />
          <Footer />
        </Switch>
      </Router>
    </Provider>
  );
});

afterEach(() => {
  wrapper.unmount();
});

describe("Find all the anchors in the Footer", () => {
  it("Footer Snapshot", () => {
    const shallowWrapper = shallow(<Footer />);
    expect(shallowWrapper).toMatchSnapshot();
  });

  it("Find three Anchors of Footer", () => {
    expect(wrapper.find(Footer).find(".grommetux-anchor")).toHaveLength(3);
  });

  it("Navigate to Status Page", () => {
    // Status page is not rendered
    expect(wrapper.find(StatusPage)).toHaveLength(0);

    // Click the anchor
    const aLink = wrapper.find(Footer).find('[href="/status"]');
    aLink.simulate("click");

    // Status Page is now rendered
    expect(wrapper.find(StatusPage)).toHaveLength(1);
  });

  it("Navigate to About Page", () => {
    // About Page is not rendered
    expect(wrapper.find(AboutPage)).toHaveLength(0);

    // Click the anchor
    const aLink = wrapper.find(Footer).find('[href="/about"]');
    aLink.simulate("click");

    // About Page is now rendered
    expect(wrapper.find(AboutPage)).toHaveLength(1);
  });
});
