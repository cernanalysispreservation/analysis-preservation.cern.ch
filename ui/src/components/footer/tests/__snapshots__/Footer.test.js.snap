// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Find all the anchors in the Footer Footer Snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <GrommetFooter />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "align": "center",
      "children": <Box
        align="center"
        announce={false}
        direction="row"
        focusable={true}
        justify="center"
        pad={
                Object {
                        "between": "medium",
                      }
        }
        responsive={true}
        tag="div"
>
        <Paragraph
                margin="none"
        >
                Copyright 
                2020
                 © CERN. Created & Hosted by CERN. Powered by Invenio Software.
        </Paragraph>
        <Menu
                closeOnClick={true}
                direction="row"
                dropAlign={
                        Object {
                                "right": "right",
                              }
                }
                dropContainer={undefined}
                pad="none"
                size="small"
        >
                <Anchor
                        animateIcon={true}
                        href="mailto:analysis-preservation-support@cern.ch"
                        label="Contact"
                        method="push"
                        tag="a"
                />
                <Anchor
                        animateIcon={true}
                        label="About"
                        method="push"
                        path="/about"
                        tag="a"
                />
                <Anchor
                        animateIcon={true}
                        label="Status"
                        method="push"
                        path="/status"
                        tag="a"
                />
        </Menu>
</Box>,
      "colorIndex": "neutral-1-a",
      "direction": "row",
      "justify": "center",
      "responsive": false,
      "size": "small",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "align": "center",
        "announce": false,
        "children": Array [
          <Paragraph
            margin="none"
>
            Copyright 
            2020
             © CERN. Created & Hosted by CERN. Powered by Invenio Software.
</Paragraph>,
          <Menu
            closeOnClick={true}
            direction="row"
            dropAlign={
                        Object {
                                    "right": "right",
                                  }
            }
            dropContainer={undefined}
            pad="none"
            size="small"
>
            <Anchor
                        animateIcon={true}
                        href="mailto:analysis-preservation-support@cern.ch"
                        label="Contact"
                        method="push"
                        tag="a"
            />
            <Anchor
                        animateIcon={true}
                        label="About"
                        method="push"
                        path="/about"
                        tag="a"
            />
            <Anchor
                        animateIcon={true}
                        label="Status"
                        method="push"
                        path="/status"
                        tag="a"
            />
</Menu>,
        ],
        "direction": "row",
        "focusable": true,
        "justify": "center",
        "pad": Object {
          "between": "medium",
        },
        "responsive": true,
        "tag": "div",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              "Copyright ",
              2020,
              " © CERN. Created & Hosted by CERN. Powered by Invenio Software.",
            ],
            "margin": "none",
          },
          "ref": null,
          "rendered": Array [
            "Copyright ",
            2020,
            " © CERN. Created & Hosted by CERN. Powered by Invenio Software.",
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Anchor
                animateIcon={true}
                href="mailto:analysis-preservation-support@cern.ch"
                label="Contact"
                method="push"
                tag="a"
/>,
              <Anchor
                animateIcon={true}
                label="About"
                method="push"
                path="/about"
                tag="a"
/>,
              <Anchor
                animateIcon={true}
                label="Status"
                method="push"
                path="/status"
                tag="a"
/>,
            ],
            "closeOnClick": true,
            "direction": "row",
            "dropAlign": Object {
              "right": "right",
            },
            "dropContainer": undefined,
            "pad": "none",
            "size": "small",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "animateIcon": true,
                "href": "mailto:analysis-preservation-support@cern.ch",
                "label": "Contact",
                "method": "push",
                "tag": "a",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "animateIcon": true,
                "label": "About",
                "method": "push",
                "path": "/about",
                "tag": "a",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "animateIcon": true,
                "label": "Status",
                "method": "push",
                "path": "/status",
                "tag": "a",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "align": "center",
        "children": <Box
          align="center"
          announce={false}
          direction="row"
          focusable={true}
          justify="center"
          pad={
                    Object {
                              "between": "medium",
                            }
          }
          responsive={true}
          tag="div"
>
          <Paragraph
                    margin="none"
          >
                    Copyright 
                    2020
                     © CERN. Created & Hosted by CERN. Powered by Invenio Software.
          </Paragraph>
          <Menu
                    closeOnClick={true}
                    direction="row"
                    dropAlign={
                              Object {
                                        "right": "right",
                                      }
                    }
                    dropContainer={undefined}
                    pad="none"
                    size="small"
          >
                    <Anchor
                              animateIcon={true}
                              href="mailto:analysis-preservation-support@cern.ch"
                              label="Contact"
                              method="push"
                              tag="a"
                    />
                    <Anchor
                              animateIcon={true}
                              label="About"
                              method="push"
                              path="/about"
                              tag="a"
                    />
                    <Anchor
                              animateIcon={true}
                              label="Status"
                              method="push"
                              path="/status"
                              tag="a"
                    />
          </Menu>
</Box>,
        "colorIndex": "neutral-1-a",
        "direction": "row",
        "justify": "center",
        "responsive": false,
        "size": "small",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "align": "center",
          "announce": false,
          "children": Array [
            <Paragraph
              margin="none"
>
              Copyright 
              2020
               © CERN. Created & Hosted by CERN. Powered by Invenio Software.
</Paragraph>,
            <Menu
              closeOnClick={true}
              direction="row"
              dropAlign={
                            Object {
                                          "right": "right",
                                        }
              }
              dropContainer={undefined}
              pad="none"
              size="small"
>
              <Anchor
                            animateIcon={true}
                            href="mailto:analysis-preservation-support@cern.ch"
                            label="Contact"
                            method="push"
                            tag="a"
              />
              <Anchor
                            animateIcon={true}
                            label="About"
                            method="push"
                            path="/about"
                            tag="a"
              />
              <Anchor
                            animateIcon={true}
                            label="Status"
                            method="push"
                            path="/status"
                            tag="a"
              />
</Menu>,
          ],
          "direction": "row",
          "focusable": true,
          "justify": "center",
          "pad": Object {
            "between": "medium",
          },
          "responsive": true,
          "tag": "div",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                "Copyright ",
                2020,
                " © CERN. Created & Hosted by CERN. Powered by Invenio Software.",
              ],
              "margin": "none",
            },
            "ref": null,
            "rendered": Array [
              "Copyright ",
              2020,
              " © CERN. Created & Hosted by CERN. Powered by Invenio Software.",
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <Anchor
                  animateIcon={true}
                  href="mailto:analysis-preservation-support@cern.ch"
                  label="Contact"
                  method="push"
                  tag="a"
/>,
                <Anchor
                  animateIcon={true}
                  label="About"
                  method="push"
                  path="/about"
                  tag="a"
/>,
                <Anchor
                  animateIcon={true}
                  label="Status"
                  method="push"
                  path="/status"
                  tag="a"
/>,
              ],
              "closeOnClick": true,
              "direction": "row",
              "dropAlign": Object {
                "right": "right",
              },
              "dropContainer": undefined,
              "pad": "none",
              "size": "small",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "animateIcon": true,
                  "href": "mailto:analysis-preservation-support@cern.ch",
                  "label": "Contact",
                  "method": "push",
                  "tag": "a",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "animateIcon": true,
                  "label": "About",
                  "method": "push",
                  "path": "/about",
                  "tag": "a",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "animateIcon": true,
                  "label": "Status",
                  "method": "push",
                  "path": "/status",
                  "tag": "a",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
